File: transport/lib/messageType/MessageError.go
package messageType

import (
	"log"
	"github.com/iain17/goTransport/transport/lib/Message"
)

type messageError struct {
	Message.Message
	Reason error `json:"reason"`
}

func init() {
	Message.Set(NewMessageError(nil))
}

func NewMessageError(reason error) *messageError {
	return &messageError{
		Message: Message.NewMessage(Message.MessageTypeError),
		Reason: reason,
	}
}

func (message messageError) Validate() error {
	return nil
}

func (message messageError) Run() error {
	log.Print(message.Reason)
	return 	nil
}

File: angular1/module/lib/MessageType/MessageError.ts
module goTransport {

    export class MessageError extends Message{
        static type = MessageType.MessageTypeError;
        private promise : Promise;

        constructor(public reason: any) {
            super(MessageError.type);
        }

        validate(): Error {
            return null;
        }

        run(): Error {
            console.error(this.reason);

            //On error
            if((this.reply instanceof MessageMethod)) {
                let promise = (this.reply as MessageMethod).getPromise();
                if(promise) {
                    console.debug(this);
                    promise.reject(this.reason);
                }
            }
            return null;
        }
    }

    MessageDefinition.set(MessageError.type, MessageError);
}